lib/Dancer2/CLI/Gen.pm

```diff
@@ -0,0 +1,322 @@
+        }
+
+        my $to_file = basename($to);
+        my $ex      = ($to_file =~ s/^\+//);
+        $to         = catfile($to_dir, $to_file) if $ex;
+
+        print "+ $to\n";
+        my $content;
+        {
+            local $/;
+            open( my $fh, '<:raw', $from ) or die "unable to open file `$from' for reading: $!";
+            $content = <$fh>;
+            close $fh;
+        }
+
+        if( $from !~ m/\.(ico|jpg|png|css|eot|map|swp|ttf|svg|woff|woff2|js)$/ ) {
+            $content = _process_template($content, $vars);
+        }
+
+        open( my $fh, '>:raw', $to ) or die "unable to open file `$to' for writing: $!";
+        print $fh $content;
+        close $fh;
+
+        if( $ex ) {
+            chmod( 0755, $to ) or warn "unable to change permissions for $to: $!";
+        }
+    }
+}
+
+sub _create_manifest {
+    my ( $self, $files, $dir ) = @_;
+
+    my $manifest_name = catfile( $dir, 'MANIFEST' );
+    open( my $manifest, '>', $manifest_name ) or die $!;
+    print $manifest "MANIFEST\n";
+
+    foreach my $file( @{ $files } ) {
+        my $filename       = substr $file->[1], length( $dir ) + 1;
+        my $basename       = basename $filename;
+        my $clean_basename = $basename;
+        $clean_basename    =~ s/^\+//;
+        $filename          =~ s/\Q$basename\E/$clean_basename/;
+        print {$manifest} "$filename\n";
+    }
+
+    close $manifest;
+}
+
+sub _add_to_manifest_skip {
+    my ( $self, $dir ) = @_;
+
+    my $filename = catfile( $dir, 'MANIFEST.SKIP' );
+    open my $fh, '>>', $filename or die $!;
+    print {$fh} "^$dir-\n";
+    close $fh;
+}
+
+sub _process_template {
+    my ( $self, $template, $tokens ) = @_;
+
+    my $engine = Dancer2::Template::Simple->new;
+    $engine->{ start_tag } = '[d2%';
+    $engine->{ stop_tag }  = '%2d]';
+    return $engine->render( \$template, $tokens );
+}
+
+# These are good candidates to move to Dancer2::CLI if other commands 
+# need them later.
+sub _get_app_path {
+    my ( $self, $path, $appname ) = @_;
+    return catdir( $path, $self->_get_dashed_name( $appname ));
+}
+
+sub _get_app_file {
+    my ( $self, $appname ) = @_;
+    $appname =~ s{::}{/}g;
+    return catfile( 'lib', "$appname.pm" );
+}
+
+sub _get_perl_interpreter {
+    return -r '/usr/bin/env' ? '#!/usr/bin/env perl' : "#!$^X";
+}
+
+sub _get_dashed_name {
+    my ( $self, $name ) = @_;
+    $name =~ s{::}{-}g;
+    return $name;
+}
+
+# Other utility methods
+sub _version_check {
+    my $self    = shift;
+    my $version = $self->parent_command->_dancer2_version;
+    return if $version =~  m/_/;
+
+    my $latest_version = 0;
+    my $resp = HTTP::Tiny->new( timeout => 5 )->get( 'https://fastapi.metacpan.org/release/Dancer2' );
+    if( $resp->{ success } ) {
+        if ( decode_json( $resp->{ content } )->{ version } =~ /(\d\.\d+)/ ) {
+            $latest_version = $1;
+        } else {
+            die "Can't understand fastapi.metacpan.org's reply.\n";
```

Comment:
Should this be a non-fatal warning?  If you're trying to scaffold a new app, it's nice to be warned if the version you have is a bit old, but it seems a bit OTT to prevent scaffolding your new app at all if we're unable to check (unless you ask not to; if this stays fatal, then this error message should probably tell you how to bypass the check).

Reply:
You are absolutely correct. I changed the way this was handled. I think it is a lot better now!